stream/stream_lavf.c

```diff
@@ -238,6 +238,109 @@ void mp_setup_av_network_options(AVDictionary **dict, const char *target_fmt,
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, "webdav"));
+        } else if (strcmp("httpproxy", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("https", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, "davs"));
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, "webdavs"));
+        } else if (strcmp("ipfs", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("ipns", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("mmsh", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, "mms"));
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, "mmshttp"));
+        } else if (strcmp("mmst", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rist", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmp", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmpe", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmps", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmpt", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmpte", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtmpts", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("rtp", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("srt", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        } else if (strcmp("srtp", ffmpeg_protos[i]) == 0) {
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, ffmpeg_protos[i]));
+        }
```

Comment:
Some macro usage could make this more readable, maybe like this:

```c
#define MATCH_PROTO(name, ...) \
    else if (!strcmp(name, ffmpeg_protos[i])) \
        a = (const char *[]) {name, #__VA_ARGS__, NULL};

const char *a[4] = {NULL};
if(0) {}
MATCH_PROTO("data")
MATCH_PROTO("http", "webdav", "dav")
// append contents of a[] to list
```

Reply:
`##__VA_ARGS__` is unfortunately gcc-specific so we'll have to live with having manually appending NULL